File: src/main/java/org/shark/example/datasource/cassandra/pojo/MessageKeyDo.java
package org.shark.example.datasource.cassandra.pojo;

import lombok.Data;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyClass;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;

@Data
@PrimaryKeyClass
public class MessageKeyDo {
    @PrimaryKeyColumn(name = "user_id", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    private Long userId;

    @PrimaryKeyColumn(name = "user_service", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    private String userService;

    @PrimaryKeyColumn(name = "item_id", ordinal = 0, type = PrimaryKeyType.CLUSTERED)
    private Long itemId;
}

File: src/main/kotlin/org/shark/example/datasource/cassandra/pojo/MessageKeyDo.kt
package org.shark.example.datasource.cassandra.pojo

import org.springframework.data.cassandra.core.cql.PrimaryKeyType
import org.springframework.data.cassandra.core.mapping.PrimaryKeyClass
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn

@PrimaryKeyClass
data class MessageKeyDo(
    @PrimaryKeyColumn(name = "user_id", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    var userId: Long? = null,

    @PrimaryKeyColumn(name = "user_service", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    var userService: String? = null,

    @PrimaryKeyColumn(name = "item_id", ordinal = 0, type = PrimaryKeyType.CLUSTERED)
    var itemId: Long? = null
)
